stout/lock.h

```diff
@@ -108,6 +109,19 @@ class Lock {
     return head_.load(std::memory_order_relaxed) == nullptr;
   }
 
+  bool IsLockedBy(Scheduler::Context* context) {
+    Waiter* current_waiter = head_.load(std::memory_order_acquire);
+    if (current_waiter != nullptr) {
```

Comment:
Can u rewrite this like `if (current_waiter == nullptr){return false;} ...`
That will improve the readability.